### PR TITLE
feat: explain dist stream plan

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).
example for the tpch-q18
```
 explain (distsql) create materialized view mv as ...
 Fragment 0
   StreamMaterialize { columns: [c_name, c_custkey, o_orderkey, o_orderdate, o_totalprice, quantity], pk_columns: [c_name, c_custkey, o_orderkey, o_orderdate, o_totalprice], order_descs: [o_totalprice, o_orderdate, c_name, c_custkey, o_orderkey] }
     StreamProject { exprs: [customer.c_name, customer.c_custkey, orders.o_orderkey, orders.o_orderdate, orders.o_totalprice, sum(lineitem.l_quantity)] }
       StreamTopN { order: "[orders.o_totalprice DESC, orders.o_orderdate ASC]", limit: 100, offset: 0 }
         StreamExchange Single from 1
 
 Fragment 1
   StreamGroupTopN { order: "[orders.o_totalprice DESC, orders.o_orderdate ASC]", limit: 100, offset: 0, group_key: [6] }
     StreamProject { exprs: [customer.c_name, customer.c_custkey, orders.o_orderkey, orders.o_orderdate, orders.o_totalprice, sum(lineitem.l_quantity), Vnode(orders.o_orderkey)] }
       StreamProject { exprs: [customer.c_name, customer.c_custkey, orders.o_orderkey, orders.o_orderdate, orders.o_totalprice, sum(lineitem.l_quantity)] }
         StreamHashAgg { group_key: [customer.c_name, customer.c_custkey, orders.o_orderkey, orders.o_orderdate, orders.o_totalprice], aggs: [count, sum(lineitem.l_quantity)] }
           StreamExchange No_shuffle from 2
 
 Fragment 2
   StreamHashJoin { type: LeftSemi, predicate: orders.o_orderkey = lineitem.l_orderkey }
     StreamExchange No_shuffle from 3
     StreamProject { exprs: [lineitem.l_orderkey] }
       StreamFilter { predicate: (sum(lineitem.l_quantity) > 1:Int32) }
         StreamProject { exprs: [lineitem.l_orderkey, sum(lineitem.l_quantity)] }
           StreamExchange No_shuffle from 8
 
 Fragment 3
   StreamHashJoin { type: Inner, predicate: orders.o_orderkey = lineitem.l_orderkey }
     StreamExchange Hash([2]) from 4
     StreamExchange Hash([0]) from 7
 
 Fragment 4
   StreamHashJoin { type: Inner, predicate: customer.c_custkey = orders.o_custkey }
     StreamExchange Hash([0]) from 5
     StreamExchange Hash([1]) from 6
 
 Fragment 5
   Chain { table: customer, columns: [c_custkey, c_name] }
     Upstream
     BatchPlanNode
 
 Fragment 6
   Chain { table: orders, columns: [o_orderkey, o_custkey, o_totalprice, o_orderdate] }
     Upstream
     BatchPlanNode
 
 Fragment 7
   Chain { table: lineitem, columns: [l_orderkey, l_quantity, l_linenumber] }
     Upstream
     BatchPlanNode
 
 Fragment 8
   StreamHashAgg { group_key: [lineitem.l_orderkey], aggs: [count, sum(lineitem.l_quantity)] }
     StreamExchange Hash([0]) from 9
 
 Fragment 9
   Chain { table: lineitem, columns: [l_orderkey, l_quantity, l_linenumber] }
     Upstream
     BatchPlanNode

(55 rows)
```
## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/stream_table_scan.rs [hunk 1]
```diff
@@ -199,7 +200,10 @@ impl StreamTableScan {
             })),
             stream_key,
             operator_id: self.base.id.0 as u64,
-            identity: format!("{}", self),
+            identity: {
+                let s = format!("{}", self);
+                s.replace("StreamTableScan", "Chain")
```
Comment (fuyufjh) [inline src/frontend/src/optimizer/plan_node/stream_table_scan.rs:205]: Ahhh 😅 I prefer to leave it as is

**Review by TennyZhuang — APPROVED** [review]

File: src/frontend/src/handler/explain.rs [hunk 2]
```diff
@@ -53,31 +55,41 @@ pub(super) fn handle_explain(
         .store(options.trace, Ordering::Release);
     // bind, plan, optimize, and serialize here
     let mut planner = Planner::new(context.into());
-    let plan = match stmt {
+    // we need to know if the plan is streaming or batch plan,
```
Comment (BugenZhao) [inline src/frontend/src/handler/explain.rs:58]: How about using `PlanNode::convention`?

File: src/frontend/src/handler/explain.rs [hunk 3]
```diff
@@ -96,57 +108,61 @@ pub(super) fn handle_explain(
...
-                    vec![PgFieldDescriptor::new(
-                        "QUERY PLAN".to_owned(),
-                        TypeOid::Varchar,
-                    )],
-                    true,
-                ));
-            }
-
-            plan
+            if options.explain_type == ExplainType::DistSQL {}
```
Comment (BugenZhao) [inline src/frontend/src/handler/explain.rs:111]:
```suggestion
            if options.explain_type == ExplainType::DistSql {}
```

File: src/frontend/src/utils/stream_graph_formatter.rs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                format!(
+                    "StreamExchange {} from {}",
+                    match dist.r#type() {
+                        risingwave_pb::stream_plan::DispatcherType::Unspecified => unreachable!(),
+                        risingwave_pb::stream_plan::DispatcherType::Hash =>
+                            format!("Hash({:?})", dist.column_indices),
+                        risingwave_pb::stream_plan::DispatcherType::Broadcast => unreachable!(),
+                        risingwave_pb::stream_plan::DispatcherType::Simple => "Single".to_string(),
+                        risingwave_pb::stream_plan::DispatcherType::NoShuffle =>
+                            "No_shuffle".to_string(),
```
Comment (BugenZhao) [inline src/frontend/src/utils/stream_graph_formatter.rs:90]:
```suggestion
                            "NoShuffle".to_string(),
```

Comment (fuyufjh) [inline src/frontend/src/utils/stream_graph_formatter.rs:90]:
```suggestion
                            "NoShuffle".to_string(),
```

File: src/frontend/src/handler/explain.rs (hunk 3 above)
Comment (fuyufjh) [inline src/frontend/src/handler/explain.rs:111]: useless?

File: src/frontend/src/optimizer/plan_node/stream_table_scan.rs (hunk 1 above)
Comment (st1page) [reply src/frontend/src/optimizer/plan_node/stream_table_scan.rs:205]: maybe It should be same as the dashboard

**Comment by chenzl25** [comment]
Should we print stream operator's internal tables?

File: src/frontend/src/optimizer/plan_node/stream_table_scan.rs (hunk 1 above)
Comment (fuyufjh) [reply src/frontend/src/optimizer/plan_node/stream_table_scan.rs:205]: So how about 

1. rename Chain to StreamTableScan in dashboard
2. or rename StreamTableScan to Chain in operator? 

I prefer 1

**Pull request merged**